src/kafka_delta_ingest.rs

```diff
@@ -0,0 +1,122 @@
+extern crate anyhow;
+
+use arrow::{
+    datatypes::Schema as ArrowSchema,
+};
+use deltalake::Schema;
+use futures::stream::StreamExt;
+use log::{info, warn};
+use rdkafka::{Message, consumer::{
+        Consumer, 
+        StreamConsumer
+    }};
+use serde_json::Value;
+use std::sync::Arc;
+use std::time::Instant;
+
+use crate::deltalake_ext;
+use crate::deltalake_ext::DeltaParquetWriter;
+
+pub async fn start(consumer: StreamConsumer, delta_table_path: &str, allowed_latency: u64, max_messages_per_batch: usize, min_bytes_per_file: usize) {
+    let mut delta_table = deltalake::open_table(delta_table_path).await.unwrap();
+    let mut json_buffer: Vec<Value> = Vec::new();
+    let mut stream = consumer.stream();
+    let metadata = delta_table.get_metadata().unwrap().clone();
+    let schema = metadata.schema;
+    let mut delta_writer = DeltaParquetWriter::for_table_path_with_schema(delta_table.table_path.clone(), &schema).await.unwrap();
+
+    let metadata = delta_table.get_metadata().unwrap().clone();
```

Comment:
aren't we getting the metadata twice here?

Reply:
Nice catch. Fixed in https://github.com/delta-io/kafka-delta-ingest/pull/1/commits/c90ff3418c7bbf0e56cce7a7dda990e3c616140a#diff-cdb6f538e826e0c2ddcbf31c3af5dd1a02b112af02245558fde629d4a3c01e6cR123.